src/main/java/edu/harvard/dbmi/avillach/dictionary/dashboarddrawer/DashboardDrawerRepository.java

```diff
@@ -0,0 +1,82 @@
+package edu.harvard.dbmi.avillach.dictionary.dashboarddrawer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+public class DashboardDrawerRepository {
+
+    private final NamedParameterJdbcTemplate template;
+
+    private static final Logger log = LoggerFactory.getLogger(DashboardDrawerRepository.class);
+
+    @Autowired
+    public DashboardDrawerRepository(NamedParameterJdbcTemplate template) {
+        this.template = template;
+    }
+
+    public List<DashboardDrawer> getDashboardDrawerRows() {
+        String materializedViewSql = """
+            select * from dictionary_db.dict.dataset_meta_materialized_view dmmv;
+            """;
+
+        String fallbackSql = """
+            SELECT d.dataset_id,
+                MAX(d.full_name) study_fullname,
+                MAX(d.abbreviation) study_abbreviation,
+                ARRAY_AGG(DISTINCT c.description) consent_groups,
+                MAX(d.description) study_summary,
+                ARRAY_AGG(DISTINCT dm.value) FILTER (where dm.key IN ('study_focus')) study_focus,
+                MAX(DISTINCT dm.value) FILTER (where dm.key IN ('study_design')) study_design,
+                MAX(DISTINCT dm.value) FILTER (where dm.key IN ('sponsor')) sponsor
+            FROM dataset d
+            JOIN dataset_meta dm ON d.dataset_id = dm.dataset_id
+            JOIN consent c ON d.dataset_id = c.dataset_id
+            GROUP BY d.dataset_id
+            """;
+
+        try {
+            return template.query(materializedViewSql, new DashboardDrawerRowMapper());
+        } catch (Exception e) {
+            log.debug("Materialized view not available, using fallback query. Error: {}", e.getMessage());
```

Comment:
Why can't we make it so this view always exists?

Reply:
https://www.postgresql.org/docs/current/rules-materializedviews.html

Materialized views are a bit different then views.  As in they are stored on disk ( in memory on some DBMS ).  They are basically a point in time view of a table.  But they need to be managed and refreshed.  

Have not use Postgres in my checkered past but they were a Saint when it came to Oracle.  Oracle has query rewrite which makes it much easier to manage.  From what I see so far in Postgres we will need to write logic blocks like this or queries will fail if Materialized view is being rewritten.

Query rewrite would fall back on the select statement used to create the materialized view if the view doesn't exist or is being refreshed.